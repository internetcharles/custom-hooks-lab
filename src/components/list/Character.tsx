import * as React from 'react';
import { Link } from 'react-router-dom';

interface AppProps {
  characters: { name: string, image: string }[]
}

const Character: React.FunctionComponent<AppProps> = ({ characters }) => {
  return (
    <ul>
      {characters.map(character => (
        <li key={character.name}>
          <Link to={`/character/${character.name}`}>
            <h1>{character.name}</h1>
          </Link>
          <img src={character.image} />
        </li>
      ))}
    </ul>
  )
}

export default Character;