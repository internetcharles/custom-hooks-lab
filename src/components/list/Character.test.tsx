import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';
import { MemoryRouter } from 'react-router';

describe('Character component', () => {
  afterEach(() => cleanup());

  it('renders Character', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Character
          characters={[{
          name: "Phil",
          image: "Phil.jpg"}]}
        />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  })
})