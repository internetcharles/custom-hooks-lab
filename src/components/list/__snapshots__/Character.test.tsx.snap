// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <a
        href="/character/Phil"
      >
        <h1>
          Phil
        </h1>
      </a>
      <img
        src="Phil.jpg"
      />
    </li>
  </ul>
</DocumentFragment>
`;
